Source/HTTP/XMLHttp/http_request_stream.cpp

```diff
@@ -60,6 +65,13 @@ HRESULT STDMETHODCALLTYPE http_request_stream::Read(
         return STG_E_READFAULT;
     }
 
+    if (m_startIndex == bodySize)
+    {
+        // Reached end of buffer
+        *pcbRead = 0;
+        return S_FALSE;
```

Comment:
is returning S_FALSE how we signal end of buffer?

Reply:
It is